src/internal/connector/discovery/discovery_test.go

```diff
@@ -197,25 +198,17 @@ func (suite *DiscoveryIntegrationSuite) TestUserInfo() {
 					path.ExchangeService: {},
 					path.OneDriveService: {},
 				},
-				HasMailBox:  true,
-				HasOneDrive: true,
 				Mailbox: api.MailboxInfo{
 					Purpose:              "user",
 					ErrGetMailBoxSetting: nil,
 				},
 			},
+			expectErr: require.NoError,
 		},
 		{
-			name: "user does not exist",
-			user: uuid.NewString(),
-			expect: &api.UserInfo{
-				DiscoveredServices: map[path.ServiceType]struct{}{},
-				HasMailBox:         false,
-				HasOneDrive:        false,
-				Mailbox: api.MailboxInfo{
-					ErrGetMailBoxSetting: api.ErrMailBoxSettingsNotFound,
-				},
-			},
+			name:      "user does not exist",
+			user:      uuid.NewString(),
+			expectErr: require.Error,
```

Comment:
are we sure the results of this test are stable? Since it got merged to start with it seems like it could have indeterminate results in some way

Reply:
For the life of me I don't see how they'd be unstable.

Reply:
and the first test run does:
```
  2023-04-20T23:58:53.016Z	INFO	mailfolders not found
      discovery_test.go:222: 
          	Error Trace:	/home/runner/work/corso/corso/src/internal/connector/discovery/discovery_test.go:222
          	            				/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.2/suite/suite.go:112
          	Error:      	An error is expected but got nil.
          	Test:       	TestDiscoveryIntegrationSuite/TestUserInfo/user_does_not_exist
          	Messages:   	<nil>
```
:stumped:

Reply:
and then, on the first test:
```
      discovery_test.go:222: 
          	Error Trace:	/home/runner/work/corso/corso/src/internal/connector/discovery/discovery_test.go:222
          	            				/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.2/suite/suite.go:112
          	Error:      	An error is expected but got nil.
          	Test:       	TestDiscoveryIntegrationSuite/TestUserInfo/user_does_not_exist
          	Messages:   	<nil>
```
:stumped: